INFRA-236: Skip installing maven artifacts

nagyo opened the pull request.

nagyo committed:
fix(releng): remove custom maven extension to skip artifact install

- do not create server product p2 repository archive
- suppress maven-assembly warnings
- bump maven-deploy and maven-install to latest version (3.1.1)
- disable javadoc build
- add bypass for issue in maven-deploy,
https://issues.apache.org/jira/browse/MINSTALL-151

nagyo committed:
Merge remote-tracking branch 'origin/7.x' into issue/INFRA-236_skip_installing_maven_artifacts

apeteri commented on dist/pom.xml:
```diff
@@ -211,80 +211,11 @@
-				</executions>
+				<artifactId>maven-deploy-plugin</artifactId>
+				<configuration>
+					<!-->This is required to bypass an issue with the maven-deploy-plugin that causes build failure in the very last plugin of a multi-module build, see https://issues.apache.org/jira/browse/MINSTALL-151 <-->
```
This XML comment format is a bit strange 🤔

apeteri reviewed (approved):
👍

cmark reviewed (approved):
🥓

nagyo committed:
fix(releng): bypass using allowIncompleteProjects flag

- any maven project with a packaging type of eclipse-repository must
have a p2 update site archive uploaded to the maven repository,
otherwise the deploy phase fails

nagyo committed:
chore(releng): build with maven 3.8.4
